### ISSUE TITLE
Flink - Add sample(s) to illustrate the use of stream cuts in a flink job

### ISSUE THREAD

**Issue by hldnova** [pravega/pravega-samples#91]
Add sample(s) to illustrate the use of stream cuts in a flink job.

**Comment by RaulGracia** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add sample to illustrate the use of stream cuts in a Flink job](https://trello.com/c/bsPc1k2l/20-add-sample-to-illustrate-the-use-of-stream-cuts-in-a-flink-job)

**Comment by hldnova** [comment]
Two simple use cases:
1. stop and resume. A flink job restarts from a saved point, resumes processing.
2. process data in a range from streams, e.g., perform auditing from last month, last year, etc.

Clients should be able to
- create streamcuts when data are ingested into pravega, e.g., create streamcuts at certain point of time.
- create streamcuts when reading through streams.

Given that Pravega does not save streamcuts, it would ideal to develop some kind of service that clients can leverage to "bookmark" streamcuts. 

Current, the pravega flink connector does not provide convenient means to manipulate stream cuts. Here is snippet from pravega `StreamCutsExample` to create a streamcut.
```
                   checkpoint = readerGroup.initiateCheckpoint(randomId + eventIndex, executor).join();
                   streamCuts.add(checkpoint.asImpl().getPositions().values().iterator().next());
```
It seems a little convoluted to use a checkpoint to get positions. And normally, flink pravega reader/writer do not deal with readergroup or checkpoint directly. 

So ideally, we should have add
- APIs in flink connector for the writer/reader to create streamcuts without knowing the details of segements, positions, checkpoints, etc., e.g., at a particular time or when interesting data appear.
- some kind of "bookmark" service that helps client to save/fetch streamcuts.

**Comment by EronWright** [comment]
I feel that the use of stream cuts to 'reset a Flink job' has a lot of overlap with checkpoints.   Checkpoints produce superior behavior because both the stream position _and the intermediate state of the processor_ are kept in sync.   I would hesitate to send mixed signals about this.

There are interesting scenarios involving Flink jobs and stream cuts.   The bookmark service is a great example, where Flink would act as an indexer or bookmark creator, and then another program (e.g. a web app or another Flink job) could use a bookmark to seek to some interesting information.

**Comment by RaulGracia** [comment]
Closed via #147.
